Revit_Core_Engine/Query/IsInRange.cs

```diff
@@ -42,9 +42,31 @@ public static bool IsInRange(this BoundingBoxXYZ box1, BoundingBoxXYZ box2, doub
         {
             if (box1 == null || box2 == null)
             {
+                BH.Engine.Base.Compute.RecordError("Intersection of the bounding boxes could not be checked. One or both of the objects are null.");
                 return false;
             }
 
+            if (box1.Transform.IsTranslation == false || box2.Transform.IsTranslation == false)
+            {
+                BH.Engine.Base.Compute.RecordError("Intersection of the bounding boxes could not be checked. One or both of the bounding boxes has unsupported transformation.");
+                return false;
+            }
+
+            if (!box1.Transform.IsIdentity)
```

Comment:
Here I would be mindful of performance, taken that this method may be run many times in geometrical search. So instead of creating a new `BoundingBoxXYZ`, I would rather do something along the lines:
```C#
{
    XYZ min1 = box1.Min;
    XYZ max1 = box1.Max;
    if (!box1.Transform.IsIdentity)
    {
        min1 += box1.Transform.Origin;
        max1 += box1.Transform.Origin;
    }
}
```